### PR TITLE
Update egg-hurtworld.json

### PR DESCRIPTION
**Pull request by Wapriaily**
# Description
- Copy the steamclient.so files to the right place
- Updated startup commands and variables

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [x] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
     ],
-    "images": [
-        "quay.io\/parkervcp\/pterodactyl-images:debian_source"
-    ],
+    "docker_images": {
+        "ghcr.io\/parkervcp\/games:source": "ghcr.io\/parkervcp\/games:source"
+    },
     "file_denylist": [],
-    "startup": ".\/Hurtworld.x86_64 -batchmode -nographics -exec \"host {{SERVER_PORT}};queryport {{QUERY_PORT}};maxplayers {{MAX_PLAYERS}};servername {{HOSTNAME}};creativemode ${CREATIVE_MODE};${ADMINS}\" -logfile $1",
+    "startup": ".\/Hurtworld.x86_64 -batchmode -nographics -exec \"host {{SERVER_PORT}};queryport {{QUERY_PORT}};maxplayers {{MAX_PLAYERS}};servername {{HOSTNAME}};addadmin {{ADMINS}}\" -logfile $1",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json:18]: Where did the creativemode mode whent to ? as the variable is still there

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
...lder is missing\r\ncd \/mnt\/server\/steamcmd\r\n\r\n# SteamCMD fails otherwise for some reason, even running as root.\r\n# This is changed at the end of the install process anyways.\r\nchown -R root:root \/mnt\r\nexport HOME=\/mnt\/server\r\n\r\n## install game using steamcmd\r\n.\/steamcmd.sh +login ${STEAM_USER} ${STEAM_PASS} ${STEAM_AUTH} +force_install_dir \/mnt\/server +app_update ${SRCDS_APPID} ${EXTRA_FLAGS} validate +quit ## other flags may be needed depending on install. looking at you cs 1.6\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\n# copy 32-bit steamclient.so\r\ncp -v linux32\/steamclient.so ..\/Hurtworld_Data\/Plugins\/x86\/steamclient.so\r\n\r\n# copy 64-bit steamclient.so\r\ncp -v linux64\/steamclient.so ..\/Hurtworld_Data\/Plugins\/x86_64\/steamclient.so\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json:27]: This is not our offical startup script and certanly not the offical steamcmd download url. change this

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json [hunk 3]
```diff
@@ -37,16 +37,18 @@
             "default_value": "405100",
             "user_viewable": true,
             "user_editable": false,
-            "rules": "required|regex:\/^(405100)$\/"
+            "rules": "required|regex:\/^(405100)$\/",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json:40]: Please use here `required|in:405100`

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json (hunk 3 above)
Comment (Wapriaily) [reply game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json]: Changed

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json (hunk 2 above)
Comment (Wapriaily) [reply game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json]: > This is not our offical startup script and certanly not the offical steamcmd download url. change this

Sorry, I replaced the link I used myself, it has been replaced

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json (hunk 1 above)
Comment (Wapriaily) [reply game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json]: > Where did the creativemode mode whent to ? as the variable is still there

Forgive me, I didn't see any variable settings about CREATIVE_MODE and ADMINS, so I changed it following the official launch command

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
     "author": "brycea@terrahost.cloud",
     "description": "Hurtworld is a hardcore multiplayer survival FPS with a focus on deep survival progression that doesn't become trivial once you establish some basic needs. Built for hardcore gamers, Hurtworld aims to punish.",
     "features": [
         "steam_disk_space"
     ],
-    "images": [
-        "quay.io\/parkervcp\/pterodactyl-images:debian_source"
-    ],
+    "docker_images": {
+        "quay.io\/parkervcp\/pterodactyl-images:debian_source": "quay.io\/parkervcp\/pterodactyl-images:debian_source"
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json:15]: Please change the image to:
`ghcr.io/parkervcp/steamcmd:debian`

The image you use is fucking old and not longer supported

**Review by gOOvER — CHANGES_REQUESTED** [review]

**Review by gOOvER — CHANGES_REQUESTED** [review]
change the image to:
`ghcr.io/parkervcp/steamcmd:debian`

quay images are since 2 yeard not supported

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
...erver\/steamcmd\r\ncd \/mnt\/server\/steamcmd\r\n\r\n# SteamCMD fails otherwise for some reason, even running as root.\r\n# This is changed at the end of the install process anyways.\r\nchown -R root:root \/mnt\r\nexport HOME=\/mnt\/server\r\n\r\n## install game using steamcmd\r\n.\/steamcmd.sh +force_install_dir \/mnt\/server +login ${STEAM_USER} ${STEAM_PASS} ${STEAM_AUTH} +app_update ${SRCDS_APPID} ${EXTRA_FLAGS} validate +quit ## other flags may be needed depending on install. looking at you cs 1.6\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\n# copy 32-bit steamclient.so\r\ncp -v linux32\/steamclient.so ..\/Hurtworld_Data\/Plugins\/x86\/steamclient.so\r\n\r\n# copy 64-bit steamclient.so\r\ncp -v linux64\/steamclient.so ..\/Hurtworld_Data\/Plugins\/x86_64\/steamclient.so\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json:27]: The whole script is old. I would suggest to update to latest steam installer script:

https://github.com/parkervcp/eggs/blob/master/scripts/steamcmd_installer.sh

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json [hunk 6]
```diff
@@ -1,30 +1,30 @@
...
             "container": "debian:buster-slim",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json:28]: please use with updated steam install script following install image:

`ghcr.io/parkervcp/installers:debian`

**Review by gOOvER — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json (hunk 6 above)
Comment (Wapriaily) [reply game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json]: done

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json (hunk 4 above)
Comment (Wapriaily) [reply game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json]: > Please change the image to: `ghcr.io/parkervcp/steamcmd:debian`
> 
> The quay image you use is fucking old and not longer supported
Changed

File: game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json (hunk 5 above)
Comment (Wapriaily) [reply game_eggs/steamcmd_servers/hurtworld/egg-hurtworld.json]: > The whole script is old. I would suggest to update to latest steam installer script:
> 
> https://github.com/parkervcp/eggs/blob/master/scripts/steamcmd_installer.sh

Updated to the latest installation script

**Comment by QuintenQVD0** [comment]
@Wapriaily You are not allowed to change the email

**Comment by Wapriaily** [comment]
> @Wapriaily You are not allowed to change the email

Sorry, I followed the tips above to remake egg

**Comment by gOOvER** [comment]
> > @Wapriaily You are not allowed to change the email
> 
> Sorry, I followed the tips above to remake egg

noone told you to change the email

**Comment by Wapriaily** [comment]
> > > @Wapriaily You are not allowed to change the email
> > 
> > 
> > Sorry, I followed the tips above to remake egg
> 
> noone told you to change the email

I remade Egg, he became my own mailbox, sorry

**Review by QuintenQVD0 — APPROVED** [review]
LGTM!

**Pull request merged**